src/main/main.js

```diff
@@ -141,20 +141,15 @@ function onReady(app) {
                 //experimentalFeatures: true,
                 disableBlinkFeatures: 'Auxclick',
                 devTools: app.config.app.debug,
-                preload: path.join(app.dir.desktopApp, 'preload.js')
+                preload: path.resolve(__dirname, 'preload.js'),
             },
-            icon: path.join(app.dir.root, 'assets/icons/png/256x256.png')
+            icon: __static + '/assets/icons/png/256x256.png'
         });
+        const webAppPath = (isDevMode())? `http://localhost:${process.env.ELECTRON_WEBPACK_WDS_PORT}/index.html` : `file://${__dirname}/index.html`;
+        
+        app.win.loadURL(webAppPath);
 
-        let webAppPath = path.join(app.dir.root, '/app/src', 'index.html');
-
-        app.win.loadURL(url.format({
-            pathname: webAppPath,
-            protocol: 'file:',
-            slashes: true
-        }));
-
-        if (app.config.app.debug) {
+        if (isDebugging()) {
```

Comment:
I wonder if we can move the Debugging / Dev mode stuff in to a single function / conditional instead of I think there are 3 now in the main process?

Reply:
@altninja I separated because dev is when we run dev mode and debugging can be for running dev tools on production as well. Thoughts?